internal/lookout/ui/src/components/lookoutV2/JobsTableRow.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from "react"
+
+import { TableRow } from "@mui/material"
+import { Row } from "@tanstack/table-core"
+import { JobTableRow, JobRow, isJobGroupRow } from "models/jobsTableModels"
+
+import { BodyCell } from "./JobsTableCell"
+import styles from "./JobsTableRow.module.css"
+
+export interface JobsTableRowProps {
+  row: Row<JobTableRow>
+  isOpenInSidebar: boolean
+  onClick?: (row: JobRow) => void
+}
+export const JobsTableRow = ({ row, isOpenInSidebar, onClick }: JobsTableRowProps) => {
+  // Helpers to avoid triggering onClick if the user is selecting text
+  const [{ pageX, pageY }, setPagePosition] = useState({ pageX: -1, pageY: -1 })
+  const isDragging = (e: React.MouseEvent) => {
+    const dragThresholdPixels = 6
+    const diffX = Math.abs(e.pageX - pageX)
+    const diffY = Math.abs(e.pageY - pageY)
+    return diffX > dragThresholdPixels || diffY > dragThresholdPixels
+  }
+
+  const original = row.original
+  const rowIsGroup = isJobGroupRow(original)
+  const rowCells = row.getVisibleCells()
+  return (
+    <TableRow
+      aria-label={row.id}
+      className={styles.rowDepthIndicator}
+      sx={{
+        backgroundSize: row.depth * 6,
+
+        ...(onClick && {
+          cursor: "pointer",
+        }),
+
+        ...(isOpenInSidebar && {
+          backgroundColor: "rgb(51, 187, 231, 0.5)",
+        }),
+
+        "&:hover": {
+          backgroundColor: "rgb(51, 187, 231, 0.2)",
+        },
+      }}
+      onMouseDown={(e) => setPagePosition(e)}
+      onClick={(e) => {
```

Comment:
Yeah, if the user drags back to within 6 pixels of the initial click location then it'll count as a "click", rather than a drag, which will toggle/change the sidebar. Didn't seem like a big deal to me though, so I was fine leaving that behaviour.